extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -562,33 +562,41 @@ class CornerstoneViewportService extends PubSubService implements IViewportServi
     if (!activeViewport) {
       console.warn('No active viewport found for', activeViewportId);
     }
-    if (activeViewport?.isReferenceViewable(metadata, { withNavigation: true })) {
+    if (activeViewport?.isReferenceViewable(metadata, WITH_NAVIGATION)) {
       return activeViewportId;
     }
 
     // Next, see if any viewport could be navigated to show the given item,
     // without considering orientation changes.
     for (const id of this.viewportsById.keys()) {
       const viewport = this.getCornerstoneViewport(id);
-      if (viewport?.isReferenceViewable(metadata, { withNavigation: true })) {
+      if (viewport?.isReferenceViewable(metadata, WITH_NAVIGATION)) {
         return id;
       }
     }
 
-    // No viewport is in the right display set/orientation to show this, so see if
-    // the active viewport could change orientations to show this
-    if (
-      activeViewport?.isReferenceViewable(metadata, { withNavigation: true, withOrientation: true })
-    ) {
-      return activeViewportId;
+    // Compute view-plane alignment scores for all viewports to prefer the one
+    // requiring the least orientation change when navigation-only is not possible.
+    const viewportAlignmentData = [];
+
+    for (const id of this.viewportsById.keys()) {
```

Comment:
Could you extract this logic into a separate function so it can be shared with the logic where there is no viewport showing that display set and we want to pick the viewport with the closest orientation?